### PR TITLE
[LC-821] update block sync logic for prevent deadlock during block_sync

### PR DESCRIPTION
**Pull request by laurenceyoon**
- fix bug in block_height_sync logic (for deadlock situation during block_sync)
- remove `role_switch_block_height` 
- add `stop_ws_heartbeat_timer`

### CODE REVIEW

File: loopchain/peer/block_manager.py [hunk 1]
```diff
@@ -656,27 +639,16 @@ def _handle_exception(e):
...
-                                                                              my_height,
-                                                                              unconfirmed_block_height,
-                                                                              max_height)
+                self.__block_request_to_peers_in_sync(peer_stubs,
+                                                      my_height,
+                                                      unconfirmed_block_height,
+                                                      max_height)
         except Exception as exc:
             _handle_exception(exc)
             return False
```
Comment (metalg0su) [inline loopchain/peer/block_manager.py]: Need to remove this return line, to ensure this func to end at the single point of truth.

Comment (laurenceyoon) [reply loopchain/peer/block_manager.py]: No problem.

File: loopchain/peer/block_manager.py [hunk 2]
```diff
@@ -578,50 +577,33 @@ def __block_request_to_peers_in_sync(self, peer_stubs, my_height, unconfirmed_bl
...
+                    elif self.blockchain.find_nid() is None:
+                        genesis_block = self.blockchain.find_block_by_height(0)
+                        self.__rebuild_nid(genesis_block)
 
                 except KeyError as e:
-                    result = False
-                    util.logger.error("fail block height sync: " + str(e))
-                    break
+                    util.logger.error(f"fail block height sync: {type(e), e}")
+                    raise
```
Comment (metalg0su) [inline loopchain/peer/block_manager.py]: Do you know when this exception could be happened?

Comment (metalg0su) [reply loopchain/peer/block_manager.py]: and.. I think more specific logging message needed. It looks very similar to log msg `except Exception as e` below

Comment (laurenceyoon) [reply loopchain/peer/block_manager.py]: No idea for what triggers KeyError, but I've added traceback.

**Review by metalg0su — APPROVED** [review]

**Review by windies21 — APPROVED** [review]

**Pull request merged**